cluster-scope/overlays/nerc-ocp-prod/feature/odf/redhatcop.redhat.io/odf-node-patcher.yaml

```diff
@@ -1,198 +1,20 @@
-          labels:
-            "cluster.ocs.openshift.io/openshift-storage": ""
----
-apiVersion: redhatcop.redhat.io/v1alpha1
-kind: Patch
-metadata:
-  name: odf-node-label-wrk-16
-  namespace: openshift-storage
-spec:
-  serviceAccountRef:
-    name: patcher
-  patches:
-    odf-node-label-wrk-16:
-      targetObjectRef:
-        apiVersion: v1
-        kind: Node
-        name: wrk-16
-      patchType: application/strategic-merge-patch+json
-      patchTemplate: |
-        metadata:
-          labels:
-            "cluster.ocs.openshift.io/openshift-storage": ""
----
-apiVersion: redhatcop.redhat.io/v1alpha1
-kind: Patch
-metadata:
-  name: odf-node-label-wrk-17
-  namespace: openshift-storage
-spec:
-  serviceAccountRef:
-    name: patcher
-  patches:
-    odf-node-label-wrk-17:
-      targetObjectRef:
-        apiVersion: v1
-        kind: Node
-        name: wrk-17
-      patchType: application/strategic-merge-patch+json
-      patchTemplate: |
-        metadata:
-          labels:
-            "cluster.ocs.openshift.io/openshift-storage": ""
----
-apiVersion: redhatcop.redhat.io/v1alpha1
-kind: Patch
-metadata:
-  name: odf-node-label-wrk-18
-  namespace: openshift-storage
-spec:
-  serviceAccountRef:
-    name: patcher
-  patches:
-    odf-node-label-wrk-18:
-      targetObjectRef:
-        apiVersion: v1
-        kind: Node
-        name: wrk-18
-      patchType: application/strategic-merge-patch+json
-      patchTemplate: |
-        metadata:
-          labels:
-            "cluster.ocs.openshift.io/openshift-storage": ""
----
-apiVersion: redhatcop.redhat.io/v1alpha1
-kind: Patch
-metadata:
-  name: odf-node-label-wrk-8
-  namespace: openshift-storage
-spec:
-  serviceAccountRef:
-    name: patcher
-  patches:
-    odf-node-label-wrk-8:
-      targetObjectRef:
-        apiVersion: v1
-        kind: Node
-        name: wrk-8
-      patchType: application/strategic-merge-patch+json
-      patchTemplate: |
-        metadata:
-          labels:
-            "cluster.ocs.openshift.io/openshift-storage": ""
----
-apiVersion: redhatcop.redhat.io/v1alpha1
-kind: Patch
-metadata:
-  name: odf-node-label-wrk-7
-  namespace: openshift-storage
-spec:
-  serviceAccountRef:
-    name: patcher
-  patches:
-    odf-node-label-wrk-7:
-      targetObjectRef:
-        apiVersion: v1
-        kind: Node
-        name: wrk-7
-      patchType: application/strategic-merge-patch+json
-      patchTemplate: |
-        metadata:
-          labels:
-            "cluster.ocs.openshift.io/openshift-storage": ""
----
-apiVersion: redhatcop.redhat.io/v1alpha1
-kind: Patch
-metadata:
-  name: odf-node-label-wrk-9
-  namespace: openshift-storage
-spec:
-  serviceAccountRef:
-    name: patcher
-  patches:
-    odf-node-label-wrk-9:
-      targetObjectRef:
-        apiVersion: v1
-        kind: Node
-        name: wrk-9
+        labelSelector:
+          matchExpressions:
+            - key: node-role.kubernetes.io/worker
+              operator: Exists
```

Comment:
I couldn't find an example in their documentation on how to do this, so this is based off of the kubernetes crd documentation and the server dry run said this was valid.

Reply:
We can run `k explain patch.spec.patches.targetObjectRef.labelSelector` to get the API docs for that field, which tell us that it's a standard label selector -- which means that it supports both `matchLabels` and `matchExpressions`:

```
$ kubectl explain patch.spec.patches.targetObjectRef.labelSelector
KIND:     Patch
VERSION:  redhatcop.redhat.io/v1alpha1

RESOURCE: labelSelector <Object>

DESCRIPTION:
     LabelSelector selects objects by label

FIELDS:
   matchExpressions     <[]Object>
     matchExpressions is a list of label selector requirements. The requirements
     are ANDed.

   matchLabels  <map[string]string>
     matchLabels is a map of {key,value} pairs. A single {key,value} in the
     matchLabels map is equivalent to an element of matchExpressions, whose key
     field is "key", the operator is "In", and the values array contains only
     "value". The requirements are ANDed.
```

So  your syntax looks correct, but we could also write:

```
labelSelector:
  matchLabels:
    node-role.kubernetes.io/worker: ""
```

Reply:
Thank you for explaining this. I was trying to read through the output of kubectl describe crd patch, but this is much more readable. I have updated it to use `matchLabels` now.